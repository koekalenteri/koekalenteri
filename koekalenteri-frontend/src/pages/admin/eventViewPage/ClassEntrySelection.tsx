import { Dispatch, SetStateAction, useCallback, useMemo } from 'react'
import { DndProvider } from 'react-dnd'
import { HTML5Backend } from 'react-dnd-html5-backend'
import { Box, Typography } from '@mui/material'
import { GridSelectionModel } from '@mui/x-data-grid'
import { Registration, RegistrationDate, RegistrationGroup } from 'koekalenteri-shared/model'
import { useRecoilState, useRecoilValue } from 'recoil'

import StyledDataGrid from '../../components/StyledDataGrid'
import { adminRegistrationIdAtom, currentEventClassRegistrationsQuery } from '../recoil'

import { DragItem } from './classEntrySelection/DragableRow'
import DragableDataGrid from './classEntrySelection/DropableDataGrid'
import { availableGroups } from './classEntrySelection/GroupColors'
import GroupHeader from './classEntrySelection/GroupHeader'
import NoRowsOverlay from './classEntrySelection/NoRowsOverlay'
import { useClassEntrySelectionColumns } from './classEntrySelection/useClassEntrySelectionColumns'

interface Props {
  eventDates?: Date[]
  setOpen?: Dispatch<SetStateAction<boolean>>
}

interface RegistrationWithGroups extends Registration {
  groups: string[]
}

export const groupKey = (rd: RegistrationDate) => rd.date.toISOString().slice(0, 10) + '-' + rd.time

const ClassEntrySelection = ({ eventDates = [], setOpen }: Props) => {
  const registrations = useRecoilValue(currentEventClassRegistrationsQuery)
  const [selectedRegistrationID, setSelectedRegistrationID] = useRecoilState(adminRegistrationIdAtom)

  const eventGroups: RegistrationGroup[] = useMemo(() => availableGroups(eventDates).map(eventDate => ({ ...eventDate, key: groupKey(eventDate), number: 0 })), [eventDates])
  const registrationsByGroup: Record<string, RegistrationWithGroups[]> = useMemo(() => {
    const byGroup: Record<string, RegistrationWithGroups[]> = { reserve: [] }
    for (const group of eventGroups) {
      byGroup[group.key] = []
    }
    for (const reg of registrations) {
      const key = reg.group?.key ?? 'reserve'
      byGroup[key] = byGroup[key] || [] // make sure the array exists
      byGroup[key].push({...reg, groups: reg.dates.map(rd => groupKey(rd))})
    }
    return byGroup
  }, [eventGroups, registrations])

  const {entryColumns, participantColumns} = useClassEntrySelectionColumns(eventDates)

  const handleDrop = (group?: RegistrationGroup) => (item: DragItem) => {
    const reg = registrations.find(r => r.id === item.id)
    if (reg && reg.group?.key !== group?.key) {
      reg.setGroup(group)
    }
    if (item.move) {
      console.log(item.id, item.move)
      delete item.move
    }
  }

  const handleSelectionModeChange = useCallback((selection: GridSelectionModel) => {
    const value = typeof selection[0] === 'string' ? selection[0] : undefined
    setSelectedRegistrationID(value)
  }, [setSelectedRegistrationID])

  const handleDoubleClick = useCallback(() => setOpen?.(true), [setOpen])

  return (
    <DndProvider backend={HTML5Backend}>
      <Typography variant='h5'>Osallistujat</Typography>
      {/* column headers only */}
      <Box sx={{ height: 40, flexShrink: 0, width: '100%', overflow: 'hidden' }}>
        <StyledDataGrid
          columns={participantColumns}
          density='compact'
          disableColumnMenu
          hideFooter
          rows={[]}
          components={{
            NoRowsOverlay: NullComponent,
          }}
        />
      </Box>
      {eventGroups.map((group) =>
        <DragableDataGrid
          key={group.key}
          group={group.key}
          columns={participantColumns}
          density='compact'
          disableColumnMenu
          hideFooter
          headerHeight={0}
          rows={registrationsByGroup[group.key] ?? []}
          onSelectionModelChange={handleSelectionModeChange}
          selectionModel={selectedRegistrationID ? [selectedRegistrationID] : []}
          components={{
            Header: GroupHeader,
            NoRowsOverlay: NoRowsOverlay,
          }}
          componentsProps={{
            header: {
              eventDates: eventDates,
              group: group,
            },
            row: {
              groupKey: group.key,
            },
          }}
          onDrop={handleDrop(group)}
        />,
      )}
      < Typography variant='h5' > Ilmoittautuneet</Typography >
      <DragableDataGrid
        columns={entryColumns}
        density='compact'
        disableColumnMenu
        flex={eventGroups.length || 1}
        rows={registrationsByGroup.reserve}
        onSelectionModelChange={handleSelectionModeChange}
        selectionModel={selectedRegistrationID ? [selectedRegistrationID] : []}
        onRowDoubleClick={handleDoubleClick}
        onDrop={handleDrop()}
      />
    </DndProvider>
  )
}

function NullComponent() {
  return null
}

export default ClassEntrySelection
