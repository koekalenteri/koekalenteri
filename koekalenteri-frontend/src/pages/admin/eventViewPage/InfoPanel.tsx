import type { EmailTemplateId, Event, Registration } from 'koekalenteri-shared/model'
import type { ChangeEvent, SyntheticEvent } from 'react'

import { useCallback, useState } from 'react'
import ExpandMore from '@mui/icons-material/ExpandMore'
import PictureAsPdfOutlined from '@mui/icons-material/PictureAsPdfOutlined'
import Accordion from '@mui/material/Accordion'
import AccordionDetails from '@mui/material/AccordionDetails'
import AccordionSummary from '@mui/material/AccordionSummary'
import Button from '@mui/material/Button'
import Link from '@mui/material/Link'
import Paper from '@mui/material/Paper'
import Tab from '@mui/material/Tab'
import Table from '@mui/material/Table'
import TableBody from '@mui/material/TableBody'
import TableCell from '@mui/material/TableCell'
import TableContainer from '@mui/material/TableContainer'
import TableHead from '@mui/material/TableHead'
import TableRow from '@mui/material/TableRow'
import Tabs from '@mui/material/Tabs'
import Typography from '@mui/material/Typography'
import { useRecoilValue } from 'recoil'

import { putInvitationAttachment } from '../../../api/event'
import useEventRegistrationInfo from '../../../hooks/useEventRegistrationsInfo'
import { API_BASE_URL } from '../../../routeConfig'
import { idTokenAtom } from '../../recoil'

interface Props {
  event: Event
  registrations: Registration[]
  onOpenMessageDialog?: (recipients: Registration[], templateId?: EmailTemplateId) => void
}

const InfoPanel = ({ event, registrations, onOpenMessageDialog }: Props) => {
  const token = useRecoilValue(idTokenAtom)
  const [attachmentKey, setAttachmentKey] = useState(event.invitationAttachment)
  const [expanded, setExpanded] = useState(true)
  const [tab, setTab] = useState(0)
  const { reserveByClass, numbersByClass, selectedByClass, stateByClass } = useEventRegistrationInfo(
    event,
    registrations
  )
  const handleExpandedChange = useCallback((event: React.SyntheticEvent, value: boolean) => {
    setExpanded(value)
  }, [])
  const handleTabChange = useCallback((e: SyntheticEvent, v: number) => {
    e.stopPropagation()
    setTab(v)
  }, [])
  const handleInvitationUpload = useCallback(
    async (e: ChangeEvent<HTMLInputElement>) => {
      if (!e.target.files) return
      const fileKey = await putInvitationAttachment(event.id, e.target.files[0], token)
      setAttachmentKey(fileKey)
    },
    [event, token]
  )

  return (
    <Accordion
      square
      disableGutters
      elevation={0}
      expanded={expanded}
      sx={{
        backgroundColor: 'background.selected',
        width: '340px',
        '& .MuiTableContainer-root': {
          backgroundColor: 'background.selected',
          width: '100%',
          p: 0,
          '& .MuiTableCell-root': { py: 0, px: 1 },
        },
      }}
      onChange={handleExpandedChange}
    >
      <AccordionSummary
        expandIcon={<ExpandMore />}
        sx={{
          minHeight: '28px',
          '& .MuiAccordionSummary-content': { m: 0.5 },
        }}
      >
        {expanded ? (
          <Tabs
            value={tab}
            onChange={handleTabChange}
            sx={{ minHeight: '24px', '& .MuiTab-root': { p: '4px 16px', minHeight: '24px', fontWeight: 'bold' } }}
          >
            <Tab label="Kokeen tilanne" />
            <Tab label="Tehtävälista" />
          </Tabs>
        ) : (
          <Typography variant="body1">Tehtäviä tekemättä: ?/?</Typography>
        )}
      </AccordionSummary>
      <AccordionDetails sx={{ p: '0px 8px 8px' }}>
        <TableContainer component={Paper} elevation={0} hidden={tab !== 0}>
          <Table>
            <TableBody>
              <TableRow>
                <TableCell colSpan={3}>
                  <Typography variant="caption" noWrap>
                    Osallistujat
                  </Typography>
                </TableCell>
              </TableRow>
              {Object.entries(numbersByClass).map(([c, nums]) => {
                return (
                  <TableRow key={c}>
                    <TableCell align="left">
                      <Typography variant="caption" noWrap fontWeight="bold" ml={2}>
                        {c}
                      </Typography>
                    </TableCell>
                    <TableCell align="right">
                      <Typography variant="caption" noWrap color={nums.invalid ? 'error' : 'info.dark'}>
                        {nums.participants} / {nums.places}
                      </Typography>
                    </TableCell>
                    <TableCell align="right">
                      <Button
                        size="small"
                        sx={{ fontSize: '0.5rem' }}
                        disabled={
                          nums.participants === 0 || nums.invalid || !['confirmed', 'picked'].includes(stateByClass[c])
                        }
                        onClick={() =>
                          onOpenMessageDialog?.(
                            selectedByClass[c],
                            stateByClass[c] === 'confirmed' ? 'picked' : 'invitation'
                          )
                        }
                      >
                        LÄHETÄ&nbsp;{stateByClass[c] === 'confirmed' ? 'KOEPAIKKAILMOITUS' : 'KOEKUTSU'}
                      </Button>
                    </TableCell>
                  </TableRow>
                )
              })}
              <TableRow>
                <TableCell colSpan={3}>
                  <Typography variant="caption" noWrap>
                    Varasijalla
                  </Typography>
                </TableCell>
              </TableRow>
              {Object.entries(numbersByClass).map(([c, nums]) => {
                return (
                  <TableRow key={c}>
                    <TableCell align="left">
                      <Typography variant="caption" noWrap fontWeight="bold" ml={2}>
                        {c}
                      </Typography>
                    </TableCell>
                    <TableCell align="right">
                      <Typography variant="caption" noWrap color="info.dark">
                        {nums.reserve}
                      </Typography>
                    </TableCell>
                    <TableCell align="right">
                      <Button
                        size="small"
                        sx={{ fontSize: '0.5rem' }}
                        disabled={nums.reserve === 0}
                        onClick={() => onOpenMessageDialog?.(reserveByClass[c], 'reserve')}
                      >
                        LÄHETÄ&nbsp;VARASIJAILMOITUS
                      </Button>
                    </TableCell>
                  </TableRow>
                )
              })}
            </TableBody>
          </Table>
        </TableContainer>
        <TableContainer component={Paper} elevation={0} hidden={tab !== 1}>
          <Table>
            <TableHead>
              <TableRow>
                <TableCell colSpan={3}>Liitteet</TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              <TableRow>
                <TableCell align="left">
                  <Typography variant="caption" noWrap fontWeight="bold" ml={2}>
                    Koekutsu
                  </Typography>
                </TableCell>
                <TableCell>
                  {attachmentKey ? (
                    <>
                      <PictureAsPdfOutlined fontSize="small" sx={{ verticalAlign: 'middle', pr: 0.5 }} />
                      <Link
                        href={`${API_BASE_URL}/file/${attachmentKey}/kutsu.pdf`}
                        rel="noopener"
                        target="_blank"
                        type="application/pdf"
                        variant="caption"
                      >
                        Kutsu.pdf
                      </Link>
                    </>
                  ) : (
                    <Typography variant="caption" fontStyle="italic">
                      Ei liitettyä tiedostoa
                    </Typography>
                  )}
                </TableCell>
                <TableCell>
                  <input
                    accept="application/pdf"
                    type="file"
                    hidden
                    id="koekutsu-file"
                    onChange={handleInvitationUpload}
                  />
                  <label htmlFor="koekutsu-file">
                    <Button component="span" size="small" sx={{ fontSize: '0.5rem' }}>
                      LIITÄ KOEKUTSU
                    </Button>
                  </label>
                </TableCell>
              </TableRow>
            </TableBody>
          </Table>
        </TableContainer>
      </AccordionDetails>
    </Accordion>
  )
}

export default InfoPanel
