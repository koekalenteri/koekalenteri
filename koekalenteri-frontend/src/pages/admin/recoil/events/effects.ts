import { Event } from 'koekalenteri-shared/model'
import { AtomEffect } from 'recoil'

import { getEvents } from '../../../../api/event'

let loaded = false

export const remoteAdminEventsEffect: AtomEffect<Event[]> = ({ setSelf, trigger }) => {
  if (trigger === 'get' && !loaded) {
    loaded = true
    getEvents().then(setSelf)
  }
}
