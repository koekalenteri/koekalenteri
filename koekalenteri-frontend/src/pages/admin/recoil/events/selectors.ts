import type { Event } from 'koekalenteri-shared/model'

import { startOfToday } from 'date-fns'
import i18next from 'i18next'
import { DefaultValue, selector, selectorFamily } from 'recoil'

import { uniqueFn } from '../../../../utils'

import {
  adminEventFilterTextAtom,
  adminEventIdAtom,
  adminEventsAtom,
  adminShowPastEventsAtom,
  newEventAtom,
} from './atoms'

export const adminEventSelector = selectorFamily<Event | undefined, string | undefined>({
  key: 'adminEventSelector',
  get:
    (eventId) =>
    ({ get }) => {
      if (!eventId) {
        return get(newEventAtom)
      }
      const events = get(adminEventsAtom)
      return events.find((e) => e.id === eventId)
    },
  set:
    (eventId) =>
    ({ get, set }, value) => {
      if (!value || value instanceof DefaultValue) {
        return
      }
      const events = get(adminEventsAtom)
      const index = events.findIndex((e) => e.id === eventId)
      const newEvents = [...events]
      const insert = index === -1
      newEvents.splice(insert ? newEvents.length : index, insert ? 0 : 1, value)
      set(adminEventsAtom, newEvents)
    },
})

export const currentAdminEventSelector = selector({
  key: 'currentAdminEvent',
  get: ({ get }) => {
    const eventId = get(adminEventIdAtom)
    return eventId ? get(adminEventSelector(eventId)) : undefined
  },
  set: ({ get, set }, newValue) => {
    if (!newValue || newValue instanceof DefaultValue) {
      return
    }
    const eventId = get(adminEventIdAtom)
    if (!eventId) {
      return
    }
    set(adminEventSelector(eventId), newValue)
  },
})

export const filteredAdminEventsSelector = selector({
  key: 'filteredAdminEvents',
  get: ({ get }) => {
    const events = get(adminEventsAtom)
    const filter = get(adminEventFilterTextAtom).toLocaleLowerCase(i18next.language)
    const showPast = get(adminShowPastEventsAtom)

    return events.filter((event) => {
      return (
        !event.deletedAt &&
        (showPast || !event.startDate || event.startDate >= startOfToday()) &&
        (!filter ||
          [event.eventType, event.name, event.location, event.official?.name, event.secretary?.name]
            .join(' ')
            .toLocaleLowerCase(i18next.language)
            .includes(filter))
      )
    })
  },
})

export const adminEventOrganizersSelector = selector({
  key: 'adminEventOrganizers',
  get: ({ get }) => {
    const events = get(adminEventsAtom)
    return uniqueFn(
      events.map((event) => event.organizer),
      (a, b) => a.id === b.id
    ).sort((a, b) => a.name.localeCompare(b.name, i18next.language))
  },
})
