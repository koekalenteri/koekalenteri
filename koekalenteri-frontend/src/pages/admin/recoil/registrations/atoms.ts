import { Registration, RegistrationGroup } from "koekalenteri-shared/model"
import { atom, atomFamily } from "recoil"

import { getRegistrations } from "../../../../api/registration"
import { logEffect, storageEffect } from "../../../recoil"


export interface RegistrationWithMutators extends Registration {
  setGroup: (group?: RegistrationGroup) => void
}

export const adminRegistrationIdAtom = atom<string | undefined>({
  key: 'adminRegistrationId',
  default: undefined,
  effects: [
    logEffect,
    storageEffect,
  ],
})

export const eventRegistrationsAtom = atomFamily<Registration[], string>({
  key: 'eventRegistrations',
  default: (eventId) => getRegistrations(eventId),
  effects: [
    logEffect,
    storageEffect,
  ],
})
