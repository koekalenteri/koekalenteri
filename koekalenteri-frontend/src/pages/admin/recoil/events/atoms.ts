import { addDays, nextSaturday, startOfDay, sub } from 'date-fns'
import { Event } from 'koekalenteri-shared/model'
import { atom, atomFamily, selector } from 'recoil'

import { uniqueClasses } from '../../../../utils'
import { logEffect, storageEffect } from '../../../recoil'

import { remoteAdminEventsEffect } from './effects'
import { adminEventSelector, currentAdminEventSelector } from './selectors'

export const adminEventsAtom = atom<Event[]>({
  key: 'adminEvents',
  default: [],
  effects: [
    logEffect,
    storageEffect,
    remoteAdminEventsEffect,
  ],
})

export const newEventStartDate = startOfDay(nextSaturday(addDays(Date.now(), 90)))
export const newEventEntryStartDate = sub(newEventStartDate, { weeks: 6 })
export const newEventEntryEndDate = sub(newEventStartDate, { weeks: 3 })

export const newEventAtom = atom<Event>({
  key: 'newEvent',
  default: {
    state: 'draft',
    startDate: newEventStartDate,
    endDate: newEventStartDate,
    entryStartDate: newEventEntryStartDate,
    entryEndDate: newEventEntryEndDate,
    classes: [],
    judges: [],
  } as unknown as Event,
  effects: [
    logEffect,
    storageEffect,
  ],
})

export const adminShowPastEventsAtom = atom<boolean>({
  key: 'adminShowPastEvents',
  default: false,
  effects: [
    logEffect,
    storageEffect,
  ],
})

export const adminEventFilterTextAtom = atom<string>({
  key: 'adminEventFilterText',
  default: '',
  effects: [
    logEffect,
    storageEffect,
  ],
})

export const adminEventIdAtom = atom<string | undefined>({
  key: 'adminEventId',
  default: undefined,
  effects: [
    logEffect,
    storageEffect,
  ],
})

export const eventClassAtom = atom<string | undefined>({
  key: 'eventClass',
  default: selector({
    key: 'eventClass/default',
    get: ({ get }) => {
      const event = get(currentAdminEventSelector)
      return uniqueClasses(event)[0] ?? event?.eventType
    },
  }),
  effects: [
    logEffect,
    storageEffect,
  ],
})

/**
 * Existing event editing, edits stored to local storage
 */
export const editableEventByIdAtom = atomFamily<Event | undefined, string>({
  key: 'editableEvent/Id',
  default: adminEventSelector,
  effects: [
    logEffect,
    storageEffect,
  ],
})
