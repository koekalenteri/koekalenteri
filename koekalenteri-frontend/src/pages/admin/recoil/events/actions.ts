import { useTranslation } from "react-i18next"
import { useNavigate } from "react-router-dom"
import { useAuthenticator } from "@aws-amplify/ui-react"
import { Event } from "koekalenteri-shared/model"
import { useSnackbar } from "notistack"
import { useRecoilState, useSetRecoilState } from "recoil"

import { putEvent } from "../../../../api/event"
import { Path } from "../../../../routeConfig"

import { adminEventIdAtom, newEventAtom } from "./atoms"
import { currentAdminEventSelector } from "./selectors"


export const useAdminEventActions = () => {
  const { user } = useAuthenticator(context => [context.user])
  const setAdminEventId = useSetRecoilState(adminEventIdAtom)
  const [currentAdminEvent, setCurrentAdminEvent] = useRecoilState(currentAdminEventSelector)
  const setNewEvent = useSetRecoilState(newEventAtom)
  const { enqueueSnackbar } = useSnackbar()
  const { t } = useTranslation()
  const navigate = useNavigate()

  return {
    copyCurrent,
    deleteCurrent,
    save,
  }

  function copyCurrent() {
    if (!currentAdminEvent) {
      return
    }
    const copy = structuredClone(currentAdminEvent)
    copy.id = ''
    copy.state = 'draft'
    copy.entries = 0
    copy.classes.forEach(c => c.entries = c.members = 0)
    delete copy.kcId

    setNewEvent(copy)
    navigate(Path.admin.newEvent)
  }

  async function save(event: Partial<Event>): Promise<Event> {
    const saved = await putEvent(event, user.getSignInUserSession()?.getIdToken().getJwtToken())
    setAdminEventId(saved.id)
    setCurrentAdminEvent(saved)
    return saved
  }

  async function deleteCurrent() {
    if (!currentAdminEvent || currentAdminEvent.deletedAt) {
      return
    }

    await save({
      ...currentAdminEvent,
      deletedAt: new Date(),
      deletedBy: user.attributes?.name || user.attributes?.email,
    })

    enqueueSnackbar(t('deleteEventComplete'), { variant: 'info' })
  }

}
