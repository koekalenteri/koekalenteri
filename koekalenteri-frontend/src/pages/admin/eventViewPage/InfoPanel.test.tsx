import { render } from '@testing-library/react'

import { eventWithEntryClosed, eventWithStaticDates } from '../../../__mockData__/events'
import { registrationsToEventWithEntryClosed } from '../../../__mockData__/registrations'

import { groupKey } from './ClassEntrySelection'
import InfoPanel from './InfoPanel'


describe('InfoPanel>', () => {
  it('renders with no registrations', () => {
    const { container } = render(<InfoPanel event={eventWithStaticDates} registrations={[]} />)

    expect(container).toMatchSnapshot()
  })

  it('renders with event with closed entry and registrations', () => {
    const { container } = render(<InfoPanel event={eventWithEntryClosed} registrations={registrationsToEventWithEntryClosed} />)

    expect(container).toMatchSnapshot()
  })

  it('renders with event with closed entry and registrations with groups', () => {
    const { container } = render(<InfoPanel event={eventWithEntryClosed} registrations={registrationsToEventWithEntryClosed.map((r, i) => ({
      ...r,
      group: {
        ...r.dates[0],
        number: i,
        key: r.cancelled ? 'cancelled' : i === 0 ? 'reserve' : groupKey(r.dates[0]),
      },
    }))} />)

    expect(container).toMatchSnapshot()
  })
})
