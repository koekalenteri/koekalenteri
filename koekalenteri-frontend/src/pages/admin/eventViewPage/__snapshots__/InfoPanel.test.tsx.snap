// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoPanel> renders with event with closed entry and registrations 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation4 MuiTableContainer-root css-13nhe7o-MuiPaper-root-MuiTableContainer-root"
  >
    <table
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            colspan="4"
            scope="col"
          >
            <b>
              Ilmoittautuneet
            </b>
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            • osallistujat
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            • varasijalla
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            4
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-10zzhxi-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              LÄHETÄ VARASIJAILMOITUS
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            • peruneet
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            1
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
             
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`InfoPanel> renders with event with closed entry and registrations with groups 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation4 MuiTableContainer-root css-13nhe7o-MuiPaper-root-MuiTableContainer-root"
  >
    <table
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            colspan="4"
            scope="col"
          >
            <b>
              Ilmoittautuneet
            </b>
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            • osallistujat
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            ALO
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            2
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-10zzhxi-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              LÄHETÄ KOEPAIKKAILMOITUS
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            AVO
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            2
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-10zzhxi-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              LÄHETÄ KOEPAIKKAILMOITUS
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            • varasijalla
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            0
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-10zzhxi-MuiButtonBase-root-MuiButton-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              LÄHETÄ VARASIJAILMOITUS
            </button>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            • peruneet
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            1
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
             
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`InfoPanel> renders with no registrations 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation4 MuiTableContainer-root css-13nhe7o-MuiPaper-root-MuiTableContainer-root"
  >
    <table
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            colspan="5"
            scope="col"
          >
            <b>
              Ilmoittautuneet
            </b>
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            • osallistujat
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            • varasijalla
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            0
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-10zzhxi-MuiButtonBase-root-MuiButton-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              LÄHETÄ VARASIJAILMOITUS
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
